velero-plugins/pvc/restore.go

```diff
@@ -0,0 +1,60 @@
+package pvc
+
+import (
+	"encoding/json"
+
+	"github.com/fusor/ocp-velero-plugin/velero-plugins/common"
+	"github.com/heptio/velero/pkg/plugin/velero"
+	"github.com/sirupsen/logrus"
+	corev1API "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// RestorePlugin is a restore item action plugin for Velero
+type RestorePlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to PVCs
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"persistentvolumeclaims"},
+	}, nil
+}
+
+// Execute action for the restore plugin for the pvc resource
+func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("[pvc-restore] Hello from PVC RestorePlugin!")
+
+	pvc := corev1API.PersistentVolumeClaim{}
+	itemMarshal, _ := json.Marshal(input.Item)
+	json.Unmarshal(itemMarshal, &pvc)
+	p.Log.Infof("[pvc-restore] pvc: %s", pvc.Name)
+
+	// Do nothing to the PVC for  a swing migration
```

Comment:
"do nothing" here seems to mean "the plugin doesn't interfere with default restore behavior" rather than "don't restore the resource" for swing migrations. I just wanted to make sure that was expected behavior vs. adding a .WithoutRestore() call to the return statement in an else clause.

Reply:
Correct. I'll make a clearer comment to that effect.